src/pages/resultsView/survival/SurvivalUtil.tsx

```diff
@@ -520,6 +531,80 @@ export function downSampling(
     });
 }
 
+export function floorScatterData(
+    scatterDataWithOpacity: ScatterData[]
+): ScatterData[] {
+    const eventInfoByMonth: { [month: number]: EventInfo } = {};
+    let eventInfo: EventInfo;
+    let result: ScatterData[] = [];
+    _.reduce(
+        scatterDataWithOpacity,
+        (eventInfoByMonth: { [month: string]: EventInfo }, item) => {
+            const month = Math.floor(item.x);
+            if (!(month in eventInfoByMonth)) {
+                // Provide initial value
+                eventInfo = {
+                    ...item,
+                    eventCount: 0,
+                    lastYInMonth: 0,
+                    censorCount: 0,
+                    lastRiskInMonth: item.atRisk!,
+                };
+            } else {
+                eventInfo = eventInfoByMonth[month];
+            }
+            eventInfoByMonth[month] = updateEventInfo(eventInfo, item);
+            return eventInfoByMonth;
+        },
+        eventInfoByMonth
+    );
+
+    result = _.map(eventInfoByMonth, (eventInfo, month) => {
+        const aggregatedScatter: ScatterData = {
+            x: parseInt(month),
+            y: eventInfo.lastYInMonth,
+            patientId: eventInfo.patientId,
+            uniquePatientKey: eventInfo.uniquePatientKey,
+            studyId: eventInfo.studyId,
+            status: eventInfo.status,
+            opacity: eventInfo.opacity,
+            group: eventInfo.group,
+            atRisk: eventInfo.lastRiskInMonth,
+            numberOfEvents: eventInfo.eventCount,
+            numberOfCensored: eventInfo.censorCount,
+        };
+        return aggregatedScatter;
+    });
+    return result;
+}
+
+function updateEventInfo(eventInfo: EventInfo, data: ScatterData): EventInfo {
```

Comment:
lets give this a more descriptive name.  like, what is the update it is performing?

Reply:
It's just updating the whole EventInfo object based on the new ScatterData, I don't know what could be a better name, so I just added some comments.